### PR TITLE
[MM-24753] Improves error message when plugin crashes during slash command

### PR DESCRIPTION
**Pull request by asimsedhain**
#### Summary

Performs a health check of the plugin before and after the slash command.
If, the plugin in unresponsive before the slash command, responds with "Plugin for the /jira command is not working. Please contact your system administrator."
If, the plugin in unresponsive after the slash command, responds with "/jira command crashed the jira plugin. Please contact your system administrator."

#### Ticket Link

Fixes: #15285, https://mattermost.atlassian.net/browse/MM-24753

### CODE REVIEW

File: plugin/environment.go [hunk 1]
```diff
@@ -473,7 +473,7 @@ func (env *Environment) RunMultiPluginHook(hookRunnerFunc func(hooks Hooks) bool
 }
 
 // performHealthCheck uses the active plugin's supervisor to verify if the plugin has crashed.
```
Comment (mickmister) [inline plugin/environment.go:475]: Nit: sense the function name changed
```suggestion
// PerformHealthCheck uses the active plugin's supervisor to verify if the plugin has crashed.
```

File: app/plugin_commands.go [hunk 2]
```diff
@@ -136,6 +137,12 @@ func (a *App) tryExecutePluginCommand(args *model.CommandArgs) (*model.Command,
 		return nil, nil, nil
 	}
 
+	// Checking if plugin is working or not
+	if err := pluginsEnvironment.PerformHealthCheck(matched.PluginId); err != nil {
+		message := fmt.Sprintf("Plugin for the /%s command is not working. Please contact your system administrator.", trigger)
+		return matched.Command, nil, model.NewAppError("ExecutePluginCommand", message, nil, "err= Plugin failed to start: "+matched.PluginId, http.StatusInternalServerError)
```
Comment (mickmister) [inline app/plugin_commands.go:143]: `Plugin failed to start` doesn't seem like the right error here. I believe this scenario is when:

- Plugin is enabled
- Plugin recently crashed and the plugin health check job has not picked up on that yet

Maybe something like `Plugin has recently crashed`?

File: app/plugin_commands.go [hunk 3]
```diff
@@ -150,5 +157,11 @@ func (a *App) tryExecutePluginCommand(args *model.CommandArgs) (*model.Command,
 	}
 
 	response, appErr := pluginHooks.ExecuteCommand(a.PluginContext(), args)
+
+	// Checking if plugin crashed after the command.
+	if err := pluginsEnvironment.PerformHealthCheck(matched.PluginId); err != nil {
+		message := fmt.Sprintf("/%s command crashed the %s plugin. Please contact your system administrator.", trigger, matched.PluginId)
+		return matched.Command, nil, model.NewAppError("ExecutePluginCommand", message, nil, "err= Plugin failed to start: "+matched.PluginId, http.StatusInternalServerError)
```
Comment (mickmister) [inline app/plugin_commands.go:164]: I believe in this scenario, this specific instance of the slash command crashed the plugin right? Maybe an error like `Plugin crashed due to slash command`, and provide some info about the command?

**Comment by asimsedhain** [comment]
Thank you for the code review, @mickmister. I have made the changes with your suggestions and also moved error messages over to `i18n/en.json` file.

I was not quite sure what to call the errors or if there was any naming convention to them. So, I have them set as `model.plugin_command_error.error.app_error` and `model.plugin_command_crash.error.app_error`. Let me know if there needs to be any changes made.

**Comment by mickmister** [comment]
@asimsedhain Do you see an opportunity to write a test for this functionality in the test files?

File: app/plugin_commands.go [hunk 4]
```diff
@@ -150,5 +156,11 @@ func (a *App) tryExecutePluginCommand(args *model.CommandArgs) (*model.Command,
 	}
 
 	response, appErr := pluginHooks.ExecuteCommand(a.PluginContext(), args)
+
+	// Checking if plugin crashed after the command.
```
Comment (ali-farooq0) [inline app/plugin_commands.go:160]:
```suggestion
	// Checking if plugin crashed after running the command
```

**Comment by asimsedhain** [comment]
> @asimsedhain Do you see an opportunity to write a test for this functionality in the test files?

I have not, @mickmister. I will take look at those.

**Comment by jasonblais** [comment]
hey @asimsedhain - let us know if you have any questions about adding a test for this functionality We'd be happy to help.

Big thanks for your contribution! :)

**Comment by asimsedhain** [comment]
Hey @jasonblais, I reached out to @mickmister for the tests. He provided me some pointers and overall has been a tremendous help.

I should be able to push an update by this weekend 😁. Thank you again for the great support 😀.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Confirmed new message is shown when plugin crashes
- Confirmed new message is shown when plugin remains in a crashed state
- Regression tested Heath Check. Seems to be working normally. 3 crashes within short amout of time leaves plugin in crashed state
- Confirmed that slash commands unrelated to a plugin are not affected so `/nothing`  still shows the original text as expected since there is no plugin to contact your admin about
LGTM!

Thanks @asimsedhain for this improvement!

Made a minor update to release testing to reflect new text.

**Comment by hanzei** [comment]
Thanks for the contribution @asimsedhain. Sorry it took so long to merge it.

**Pull request merged**